src/mesh/mesh.cpp

```diff
@@ -601,6 +653,24 @@ Mesh::Mesh(ParameterInput *pin, IOWrapper &resfile, Properties_t &properties,
     PARTHENON_FAIL(msg);
   }
 
+  // the following two error checks in combination with the driver task list logic
+  // in driver.hpp are a workaround to make scratch pad allocation work in a
+  // multi host thread environment
+  if (num_mesh_threads_ > num_mesh_streams_) {
+    msg << "### FATAL ERROR in Mesh constructor" << std::endl
+        << "Number of threads (" << num_mesh_threads_ << ") must be <= num_streams ("
+        << num_mesh_streams_ << ")" << std::endl;
+    PARTHENON_FAIL(msg);
```

Comment:
Should this not be `PARTHENON_FAIL(msg.str().c_str())`? I'm surprised it works otherwise.

Reply:
Andrew introduce a overload to that function some time ago that converts it transparently.

Reply:
Ah i see that now!